src/roman_datamodels/datamodels.py

```diff
@@ -405,7 +405,7 @@ def __init__(self, init=None, iscopy=False, memmap=False, return_open=True, save
         if init is None:
             # don't populate container
             pass
-        elif isinstance(init, list):
+        elif isinstance(init, (list, Sequence)):
```

Comment:
```suggestion
        elif isinstance(init, Iterable):
```

This will allow it to handle anything that follows the iterable protocol, the later checks will cover off any cases where this does not make sense.

Note `Iterable` is a part of the `abc.collections` builtin Python module, so line 16 will need to be updated in addition to this change in order for it to work.